SpatialGDK/Source/SpatialGDKFunctionalTests/Private/TestMaps/GeneratedTestMap.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestMaps/GeneratedTestMap.h"
+#include "Engine/ExponentialHeightFog.h"
+#include "Engine/SkyLight.h"
+#include "Engine/StaticMeshActor.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "FileHelpers.h"
+#include "GameFramework/PlayerStart.h"
+#include "Tests/AutomationEditorCommon.h"
+#include "UObject/ConstructorHelpers.h"
+
+UGeneratedTestMap::UGeneratedTestMap()
+{
+	ConstructorHelpers::FObjectFinder<UStaticMesh> PlaneAsset(TEXT("StaticMesh'/Engine/BasicShapes/Plane.Plane'"));
+	check(PlaneAsset.Succeeded());
+	PlaneStaticMesh = PlaneAsset.Object;
+
+	ConstructorHelpers::FObjectFinder<UMaterial> MaterialAsset(TEXT("Material'/Engine/BasicShapes/BasicShapeMaterial.BasicShapeMaterial'"));
+	check(MaterialAsset.Succeeded());
+	BasicShapeMaterial = MaterialAsset.Object;
+
+	MapCategory = CI_FAST; // We will set the default to CI_FAST, even if almost every map will override this anyway
+						   // Just in case we can make a newly written map be immediately run in CI when the PR is still in development, so
+						   // that the developer gets feedback about whether their new map and presumably new test passes in CI
+}
+
+void UGeneratedTestMap::GenerateMap()
+{
+	GenerateBaseMap();
+	CreateCustomContentForMap();
+	SaveMap();
+}
+
+FString UGeneratedTestMap::GetGeneratedMapFolder()
+{
+	return FPaths::ConvertRelativePathToFull(FPaths::ProjectContentDir() + TEXT("Intermediate/Maps/"));
+}
+
+void UGeneratedTestMap::GenerateBaseMap()
+{
+	UWorld* CurrentWorld = FAutomationEditorCommonUtils::CreateNewMap();
+	ULevel* CurrentLevel = CurrentWorld->GetCurrentLevel();
+
+	AStaticMeshActor* Plane =
+		CastChecked<AStaticMeshActor>(GEditor->AddActor(CurrentLevel, AStaticMeshActor::StaticClass(), FTransform::Identity));
+	Plane->GetStaticMeshComponent()->SetStaticMesh(PlaneStaticMesh);
+	Plane->GetStaticMeshComponent()->SetMaterial(0, BasicShapeMaterial);
+	Plane->SetActorScale3D(FVector(
+		500, 500, 1)); // Make the initial platform larger so things don't fall off for tests which use a large area (visibility test)
+
+	AExponentialHeightFog* Fog =
+		CastChecked<AExponentialHeightFog>(GEditor->AddActor(CurrentLevel, AExponentialHeightFog::StaticClass(), FTransform::Identity));
+
+	APlayerStart* PlayerStart =
+		CastChecked<APlayerStart>(GEditor->AddActor(CurrentLevel, APlayerStart::StaticClass(), FTransform(FVector(-500, 0, 100))));
```

Comment:
```suggestion
		CastChecked<APlayerStart>(GEditor->AddActor(CurrentLevel, APlayerStart::StaticClass(), FTransform(StartingPoint)));
```
Put a constant, add comment about LBs, maybe start in one specific quadrant by default.